modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationServiceImpl.java

```diff
@@ -529,4 +533,55 @@ protected void atomic(String id, Consumer<KeyValueStore> consumer) {
         throw new RuntimeException("Cannot lock for write migration: " + id + ", already locked: " + currentLock);
     }
 
+    @Override
+    public Migration getMigration(String id) {
+        MigrationDescriptor descriptor = getDescriptor(XP_CONFIG, id);
+        return new Migration(id, descriptor.getDescription(), descriptor.getDescriptionLabel(), getStatus(id),
+                descriptor.getSteps());
```

Comment:
The steps should be filtered to only return the available steps for the given migration, like what is done on the JSF UI (see https://github.com/nuxeo/nuxeo-jsf-ui/blob/master/code/nuxeo-admin-center-jsf/src/main/java/org/nuxeo/ecm/admin/MigrationAdminBean.java#L49 for instance).

Reply:
I made the `getAvailableSteps` return a map so I can give it here in the constructor and leverage `getAvailableSteps`